x/valset-pref/msg_server_test.go

```diff
@@ -124,3 +124,82 @@ func (suite *KeeperTestSuite) TestSetValidatorSetPreference() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestDelegateToValidatorSet() {
+	suite.SetupTest()
+
+	// prepare validators to delegate to
+	preferences := suite.PrepareDelegateToValidatorSet()
+
+	tests := []struct {
+		name           string
+		delegator      sdk.AccAddress
+		coin           sdk.Coin
+		expectedShares []sdk.Dec
+		expectPass     bool
+		valSetExists   bool
+	}{
+		{
+			name:           "Delegate to valid validators!",
+			delegator:      sdk.AccAddress([]byte("addr1---------------")),
+			coin:           sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(10_000_000)),                      // amount to delegate
+			expectedShares: []sdk.Dec{sdk.NewDec(5_000_000), sdk.NewDec(3_000_000), sdk.NewDec(2_000_000)}, // expected shares after delegation
```

Comment:
Just fyi this abstraction made me read it twice before I got what was happening. Maybe in the comments just explicitly state in the comments that test's val set is defined as 50%,30%,and 20% instead of inferring it :)